plugins/unicodelib/unicodelib.cpp

```diff
@@ -719,6 +720,34 @@ unsigned unicodeEditDistanceV4(UnicodeString & left, UnicodeString & right, unsi
     return da[mask(leftLen-1)][rightLen-1];
 }
 
+void translate(UnicodeString & source, UChar const * sear, unsigned searLen, UChar const * repl, unsigned replLen)
+{
+    UnicodeString search(sear, searLen);
+    UnicodeString replace(repl, replLen);
+    if (search.countChar32() != replace.countChar32() || source.isEmpty() || search.isEmpty() || replace.isEmpty())
+    {
+        return;
+    }
+    StringCharacterIterator it(source);
+    source.remove();
+    int32_t Beg = it.setToStart();
+    while (Beg != it.endIndex())
```

Comment:
Without studying the details of CharacterIterator, I cannot tell if this is a problem.  However, it is generally a good practice to use the corresponding pairs:  setToStart() with setToEnd(); or startIndex() with endIndex().  There are sometimes subtle differences.  If you are counting on these differences, you should call that out in a comment.

One minor quibble is that the variable you are setting and updating appears to me to really be a position and not the beginning.